### PR TITLE
Add rotation center point to RotateTransform

### LINKED ISSUE #4149: Add the rotation center point in RotateTransform
Hello!

In WPF we can add the [specified point in `RotateTransform`](https://github.com/dotnet/wpf/blob/ac9d1b7a6b0ee7c44fd2875a1174b820b3940619/src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/RotateTransform.cs#L49-L60). Also, we can do it [in UWP](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.media.rotatetransform).

I correctly understood that in Avalonia we can do this only by creating a `TransformGroup` with `TranslateTransform`, `RotateTransform` elements?

### PR DESCRIPTION
**Pull request by rstm-sf**
## What does the pull request do?
Add rotation center point to RotateTransform

## What is the current behavior?
Now we do not have the ability to set the center of rotation

## What is the updated/expected behavior with this PR?
We can set the center of rotation

## How was the solution implemented (if it's not obvious)?
Added page for RotateTransform to ControlCatalog

## Checklist

- [x] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

## Fixed issues
Fixes #4149

### CODE REVIEW

File: samples/ControlCatalog/Models/MathConverter.cs [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+using Avalonia.Data.Converters;
+
+namespace ControlCatalog.Models
+{
+    public class MathConverter : IValueConverter
+    {
+        private static readonly List<char> _allOperators = new List<char> { '+', '-', '*', '/', '%', '(', ')' };
+
+        private static readonly List<string> _grouping = new List<string> { "(", ")" };
+        private static readonly List<string> _operators = new List<string> { "+", "-", "*", "/", "%" };
```
Comment (KvanTTT) [inline samples/ControlCatalog/Models/MathConverter.cs:13]: It's better to use arrays instead of lists here. It eliminates excess `.ToArray()` calls.

**Review by KvanTTT — COMMENTED** [review]
I think it can be rewritten in more performance and zero-allocation way. I try to do it after this PR will be merged.

File: samples/ControlCatalog/Models/MathConverter.cs (hunk 1 above)
Comment (rstm-sf) [reply samples/ControlCatalog/Models/MathConverter.cs]: Thank you for your attention!

I decided to submit negative values ​​and this code stopped working. Therefore, I decided to make a simple option

File: samples/ControlCatalog/MainView.xaml [hunk 2]
```diff
@@ -60,6 +60,7 @@
       <TabItem Header="TextBlock"><pages:TextBlockPage/></TabItem>
       <TabItem Header="ToggleSwitch"><pages:ToggleSwitchPage/></TabItem>
       <TabItem Header="ToolTip"><pages:ToolTipPage/></TabItem>
+      <TabItem Header="Transform"><pages:TransformPage/></TabItem>
```
Comment (rstm-sf) [inline samples/ControlCatalog/MainView.xaml:63]: I understand that the TransformPage in the directory does not look very good. But then where is it better to move it or just delete it?

**Comment by rstm-sf** [comment]
Ok. I thought here that this change will have a bad effect on performance when we have center point == default. Therefore, it is possible that this change is not worth applying

File: samples/RenderDemo/Pages/RenderTargetBitmapPage.cs [hunk 3]
```diff
@@ -31,9 +31,7 @@ public override void Render(DrawingContext context)
         {
             using (var ctxi = _bitmap.CreateDrawingContext(null))
             using(var ctx = new DrawingContext(ctxi, false))
-            using (ctx.PushPostTransform(Matrix.CreateTranslation(-100, -100)
-                                         * Matrix.CreateRotation(_st.Elapsed.TotalSeconds)
-                                         * Matrix.CreateTranslation(100, 100)))
+            using (ctx.PushPostTransform(Matrix.CreateRotation(_st.Elapsed.TotalSeconds, 100, 100)))
```
Comment (rstm-sf) [inline samples/RenderDemo/Pages/RenderTargetBitmapPage.cs:34]: But I think this change is good

**Comment by grokys** [comment]
Thing is, there's already [`RenderTransformOrigin`](https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Visuals/Visual.cs#L228-L235) which is used to set the rotation center - this works the same in [WPF](https://docs.microsoft.com/en-us/dotnet/api/system.windows.uielement.rendertransformorigin?view=netcore-3.1) and [UWP](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.rendertransformorigin?view=winrt-19041) so I'm not sure this is necessary?

**Comment by rstm-sf** [comment]
Thanks for the clarification!

It’s more of an optimization when offset is needed -- the number of arithmetic operations is reduced. However, I do not know how often such a case is used. Therefore, I think it’s better to leave this thinking to the future when there are users who need it.

**Pull request closed without merging**